### ISSUE TITLE
(4.6.4) Different value between file and read

### ISSUE THREAD

**Issue by r0perice** [Unidata/thredds#539]
Hi,

I'm working on a generic reader and writer in Java for Netcdf4 files. I'm using the 4.6.4 library. I was using [nccmp](http://nccmp.sourceforge.net/#overview) and I noticed some strange things.

In my input netcdf4 file I have a Variable like that :
```
double swath_earth_start_time(number_of_swaths=1);
  :units = "seconds since 2000-01-01 00:00:00.0";
  :_FillValue = 9.969209968386869E36; // double
  :tai_utc_difference = "<value_from_file>";
  :leap_second = "leap_second";
  :long_name = "Time of first Earth measurement in the swath";
```
In **nccmp** when I compare my input file with my output file I see :

![swath-earth-start-time](https://cloud.githubusercontent.com/assets/7713432/14856899/492f9c20-0c9a-11e6-82d4-3a5286de14e3.png)

I used [Data Explorer](https://github.com/pedro-vicente/data-explorer) to see my Variable data value :

![data-explorer](https://cloud.githubusercontent.com/assets/7713432/14857334/2c6e990e-0c9c-11e6-873f-de1c06a0aabe.png)

In my code I toggled a breakpoint the inspect to read.

![breakpoint](https://cloud.githubusercontent.com/assets/7713432/14857293/0761229e-0c9c-11e6-9f08-b6dee7e1c8e8.png)

![read](https://cloud.githubusercontent.com/assets/7713432/14857300/1041a460-0c9c-11e6-9cea-6c0df2c2e3bb.png)

Using the **toolsUI-4.6.4.jar** I see :

![toolui](https://cloud.githubusercontent.com/assets/7713432/14857349/410d8da2-0c9c-11e6-8ca2-2beeea832fe6.png)

I don't know who to trust here :confused: 

Any help appreciated.

Regards,

Robin

**Comment by DennisHeimbigner** [comment]
Can you give the full path (or url) for that file?

**Comment by cwardgar** [comment]
I would consider ncdump to be authoritative in these instances. But yes, can you provide the file? We may have a bug.

**Comment by r0perice** [comment]
Hi,

I'm using public EUMETSAT dataset from http://www.eumetsat.int/website/home/News/DAT_2711905.html?lang=EN.

FTP : ftp://ftp.eumetsat.int/pub/OPS/out/test-data/FCI_L1C_Format_Familiarisation

The file I used is : FDHSI Files > Body Chunks > W_XX‐EUMETSAT‐Darmstadt,IMG+SAT,MTI1+FCI‐1C‐RRAD‐FDHSI‐FD‐x‐CHK‐BODY‐x‐L2P‐NC4E‐C‐EUMT_20130804120000_GTT_DEV_20130804120000_20130804120007_N_x_C_0003.nc.

Regards.

Robin

**Comment by cwardgar** [comment]
Okay, I grabbed the file and `ncdump`ed the variable:
```
ncdump -v /data/vis_04/swath/swath_earth_start_time gh-539.nc
```
```
      data:

       swath_earth_start_time = _ ;
```
Where:
```
       ncdump uses `_' to represent data values that are equal to the `_FillValue' attribute for  a  vari-
       able,  intended to represent data that has not yet been written.  If a variable has no `_FillValue'
       attribute, the default fill value for the variable type is used unless  the  variable  is  of  byte
       type.
```
That's the expected result. I'll look into why we're getting `0.0` in ToolsUI.

**Comment by cwardgar** [comment]
So I tracked down the [source of the problem](https://github.com/Unidata/thredds/blob/master/cdm/src/main/java/ucar/nc2/iosp/hdf5/H5header.java#L76). Setting the flag to `true` causes the expected value to be reported.

@DennisHeimbigner This was introduced in 1bdf61d2aa2b7fd92b4cc60f4274664d3e0deae5. Can you provide any info about this? Do we still need the flag?

**Comment by DennisHeimbigner** [comment]
Does it work if you set HDF5FILL to true?

**Comment by cwardgar** [comment]
Yes.

**Comment by DennisHeimbigner** [comment]
Since I have no idea any more, remove it.

**Comment by r0perice** [comment]
Hi,

Thanks for you quick response !
 Is the fix going to be available for v4.6.4 or v4.6.5 or I need to update my code to v5.0.0 ? 

Regards.

Robin

**Comment by dopplershift** [comment]
The fix for this will end up in a 4.6.x release.

**Comment by cwardgar** [comment]
Fixed in #544. Thanks Robin!
